lib/expired_record_cleaner.rb

```diff
@@ -0,0 +1,53 @@
+class ExpiredRecordCleaner
+  attr_reader :verbose, :fiscal_year_start
+
+  def initialize(dtim, ok_to_act = false, verbose = false)
+    @fiscal_year_start = self.class.fiscal_year_start(dtim)
+    @ok_to_act = ok_to_act
+    @verbose = verbose
+  end
+
+  def vacuum_old_proposals
+    proposals = Proposal.pending.where("created_at < '#{@fiscal_year_start}'")
+    ids = []
+    proposals.each do |proposal|
+      vacuum_proposal(proposal)
+      ids << proposal.id
+    end
+    ids
+  end
+
+  def vacuum_proposal(proposal)
+    if !proposal.requester
+      handle_no_requester(proposal)
+    else
+      handle_cancellation(proposal)
+    end
+  end
+
+  private
+
+  def self.fiscal_year_start(dtim)
+    year = (dtim.month >= 10) ? dtim.year : dtim.year - 1
+    Time.zone.parse("#{year}-10-01")
+  end
+
+  def send_reminder_email(proposal)
+    CommunicartMailer.proposal_cancellation(proposal).deliver_later
+  end
+
+  def handle_no_requester(proposal)
+    @verbose and STDERR.puts "#{proposal.id} <= no Requester defined"
```

Comment:
I think what this is doing (if I understand it right) would be clearer if there were an `if`

```
if @verbose
  STDERR.puts "#{proposal.id} <= no Requester defined"
end
```

same for case below